app/src/main/java/com/example/eric/pokedex/MainActivity.java

```diff
@@ -0,0 +1,290 @@
+                                recyclerView.setAdapter(pokedexAdapter);
+                                recyclerView.scrollToPosition(0);
+                                Log.i("Type ", item);
+                            } catch (Exception e){
+                                Log.i("List Count", Integer.toString(mDrawerList.getAdapter().getCount()));
+                            }
+
+
+                        }
+
+                    }
+                }
+            }
+        });
+
+
+        pokedex = new Pokedex();
+
+        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+
+        pokedexAdapter = new PokedexAdapter(getApplicationContext(), pokedex.getPokemon(), new Comparator<Pokedex.Pokemon>() {
+            @Override
+            public int compare(Pokedex.Pokemon pokemon, Pokedex.Pokemon t1) {
+                return pokemon.name.compareTo(t1.name);
+            }
+        });
+        copyPokemons = pokedexAdapter.pokemons;
+
+        recyclerView.setAdapter(pokedexAdapter);
+
+        mModels = new ArrayList<>();
+        for (Pokedex.Pokemon k: pokedex.getPokemon()) {
+            mModels.add(k);
+        }
+        pokedexAdapter.add(mModels);
+
+        filterReturn();
+    }
+
+    public void filterReturn(){
+        if (code == 1) {
+            int[]stats = new int[3];
+            stats[0] = minHp;
+            stats[1] = minAttack;
+            stats[2] = minDefense;
+            ArrayList<Pokedex.Pokemon> filteredModelList = filterStats(mModels, stats);
+            pokedexAdapter.pokemons = filteredModelList;
+            recyclerView.setAdapter(pokedexAdapter);
+            this.recyclerView.scrollToPosition(0);
+
+        } else if (code == 2){
+            ArrayList<Pokedex.Pokemon> filteredModelList = filterRandom(mModels, random);
+            pokedexAdapter.pokemons = filteredModelList;
+            recyclerView.setAdapter(pokedexAdapter);
+            this.recyclerView.scrollToPosition(0);
+        }
+
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.searchbar, menu);
+
+        SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
+        SearchView searchView = (SearchView) menu.findItem(R.id.action_search).getActionView();
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(getComponentName()));
+        searchView.setOnQueryTextListener(this);
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query){
+        try {
+            int x = Integer.parseInt(query);
+            final ArrayList<Pokedex.Pokemon> filteredModelList = filterNum(mModels, x);
+            pokedexAdapter.pokemons = filteredModelList;
+            recyclerView.setAdapter(pokedexAdapter);
+            this.recyclerView.scrollToPosition(0);
+            return true;
+
+        } catch (Throwable e) {
```

Comment:
why use error handling as an if statement?? Now if there's some unrelated error, you'll execute this block